{"id":null,"jsonrpc":"2.0","method":"textDocument/extendPublishDiagnostics","params":{"diagnostics":[{"category":74,"code":74,"message":"undeclared type name 'RefArray'","range":{"end":{"character":20,"line":1},"start":{"character":19,"line":1}},"severity":1,"source":"Cangjie"},{"category":75,"code":75,"message":"undeclared identifier 'RefArray'","range":{"end":{"character":38,"line":1},"start":{"character":37,"line":1}},"severity":1,"source":"Cangjie"},{"category":74,"code":74,"message":"undeclared type name 'RefArray'","range":{"end":{"character":20,"line":9},"start":{"character":19,"line":9}},"severity":1,"source":"Cangjie"},{"category":75,"code":75,"message":"undeclared identifier 'RefArray'","range":{"end":{"character":38,"line":9},"start":{"character":37,"line":9}},"severity":1,"source":"Cangjie"},{"category":727,"code":727,"message":"'getMessage' is not a member of class 'Error'","range":{"end":{"character":49,"line":13},"start":{"character":39,"line":13}},"severity":1,"source":"Cangjie"},{"category":74,"code":74,"message":"undeclared type name 'RefArray'","range":{"end":{"character":20,"line":17},"start":{"character":19,"line":17}},"severity":1,"source":"Cangjie"},{"category":75,"code":75,"message":"undeclared identifier 'RefArray'","range":{"end":{"character":38,"line":17},"start":{"character":37,"line":17}},"severity":1,"source":"Cangjie"},{"category":186,"code":186,"message":"the exception catch type must be class and extends from core.Exception or core.Error","range":{"end":{"character":18,"line":20},"start":{"character":17,"line":20}},"severity":1,"source":"Cangjie"},{"category":74,"code":74,"message":"undeclared type name 'Throwable'","range":{"end":{"character":18,"line":20},"start":{"character":17,"line":20}},"severity":1,"source":"Cangjie"},{"category":74,"code":74,"message":"undeclared type name 'RefArray'","range":{"end":{"character":20,"line":25},"start":{"character":19,"line":25}},"severity":1,"source":"Cangjie"},{"category":75,"code":75,"message":"undeclared identifier 'RefArray'","range":{"end":{"character":38,"line":25},"start":{"character":37,"line":25}},"severity":1,"source":"Cangjie"},{"category":727,"code":727,"message":"'getMessage' is not a member of class 'Exception'","range":{"end":{"character":49,"line":29},"start":{"character":39,"line":29}},"severity":1,"source":"Cangjie"},{"category":588,"code":588,"message":"'main' declaration doesn't need 'func' keyword","range":{"end":{"character":9,"line":33},"start":{"character":5,"line":33}},"severity":1,"source":"Cangjie"},{"category":186,"code":186,"message":"the exception catch type must be class and extends from core.Exception or core.Error","range":{"end":{"character":17,"line":36},"start":{"character":16,"line":36}},"severity":1,"source":"Cangjie"},{"category":74,"code":74,"message":"undeclared type name 'RuntimeException'","range":{"end":{"character":17,"line":36},"start":{"character":16,"line":36}},"severity":1,"source":"Cangjie"},{"category":727,"code":727,"message":"'getMessage' is not a member of class 'Father'","range":{"end":{"character":26,"line":76},"start":{"character":16,"line":76}},"severity":1,"source":"Cangjie"},{"category":186,"code":186,"message":"the exception catch type must be class and extends from core.Exception or core.Error","range":{"end":{"character":17,"line":85},"start":{"character":16,"line":85}},"severity":1,"source":"Cangjie"},{"category":74,"code":74,"message":"undeclared type name 'RuntimeException'","range":{"end":{"character":17,"line":85},"start":{"character":16,"line":85}},"severity":1,"source":"Cangjie"},{"category":186,"code":186,"message":"the exception catch type must be class and extends from core.Exception or core.Error","range":{"end":{"character":17,"line":96},"start":{"character":16,"line":96}},"severity":1,"source":"Cangjie"},{"category":74,"code":74,"message":"undeclared type name 'RuntimeException'","range":{"end":{"character":17,"line":96},"start":{"character":16,"line":96}},"severity":1,"source":"Cangjie"},{"category":74,"code":74,"message":"undeclared type name 'RefArray'","range":{"end":{"character":20,"line":106},"start":{"character":19,"line":106}},"severity":1,"source":"Cangjie"},{"category":75,"code":75,"message":"undeclared identifier 'RefArray'","range":{"end":{"character":38,"line":106},"start":{"character":37,"line":106}},"severity":1,"source":"Cangjie"},{"category":727,"code":727,"message":"'getMessage' is not a member of class 'ArithmeticException'","range":{"end":{"character":64,"line":111},"start":{"character":54,"line":111}},"severity":1,"source":"Cangjie"},{"category":74,"code":74,"message":"undeclared type name 'Throwable'","range":{"end":{"character":46,"line":117},"start":{"character":45,"line":117}},"severity":1,"source":"Cangjie"},{"category":74,"code":74,"message":"undeclared type name 'Throwable'","range":{"end":{"character":26,"line":120},"start":{"character":25,"line":120}},"severity":1,"source":"Cangjie"},{"category":74,"code":74,"message":"undeclared type name 'Throwable'","range":{"end":{"character":46,"line":120},"start":{"character":45,"line":120}},"severity":1,"source":"Cangjie"},{"category":74,"code":74,"message":"undeclared type name 'Throwable'","range":{"end":{"character":27,"line":121},"start":{"character":26,"line":121}},"severity":1,"source":"Cangjie"},{"category":183,"code":183,"message":"generic type should be used with type argument","range":{"end":{"character":40,"line":121},"start":{"character":39,"line":121}},"severity":1,"source":"Cangjie"},{"category":74,"code":74,"message":"undeclared type name 'RefArray'","range":{"end":{"character":20,"line":123},"start":{"character":19,"line":123}},"severity":1,"source":"Cangjie"},{"category":75,"code":75,"message":"undeclared identifier 'RefArray'","range":{"end":{"character":38,"line":123},"start":{"character":37,"line":123}},"severity":1,"source":"Cangjie"},{"category":727,"code":727,"message":"'getMessage' is not a member of class 'Exception'","range":{"end":{"character":47,"line":127},"start":{"character":37,"line":127}},"severity":1,"source":"Cangjie"},{"category":727,"code":727,"message":"'getMessage' is not a member of class 'Exception'","range":{"end":{"character":110,"line":129},"start":{"character":100,"line":129}},"severity":1,"source":"Cangjie"}],"uri":"file://***//lsp/test/testChr/normalTest/Exception/src/exception.cj"}}
